flowredux/src/commonMain/kotlin/com/freeletics/flowredux/dsl/FlowReduxStoreBuilder.kt

```diff
@@ -1,20 +1,14 @@
 package com.freeletics.flowredux.dsl
 
-import com.freeletics.flowredux.SideEffect
+import com.freeletics.flowredux.sideeffects.InStateSideEffectBuilder
 import kotlin.reflect.KClass
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 
 @ExperimentalCoroutinesApi
 @FlowReduxDsl
-public class FlowReduxStoreBuilder<S : Any, A : Any> {
+public class FlowReduxStoreBuilder<S : Any, A : Any> internal constructor() {
 
-    private val builderBlocks: MutableList<InStateBuilderBlock<*, S, A>> = ArrayList()
-
-    internal fun generateSideEffects(): List<SideEffect<S, A>> {
-        return builderBlocks.flatMap { builder ->
-            builder.generateSideEffects()
-        }
-    }
+    internal val sideEffectBuilders: MutableList<InStateSideEffectBuilder<out S, S, A>> = ArrayList()
```

Comment:
Nitpicking: Should this really be mutable List from a public API (although internal modifier) point of view

Reply:
Was a copy paste mistake from `BaseBuilderBlock`. There it needs to be exposed because subclasses need to be able to add something and `protected` gets flag for exposing an `internal` type.